File: DataSets/DataSet-New/LeetCodeTasks - Debug/WithBugs/3101_count-alternating-subarrays/solutions/cpp/solution.cpp
#include <vector>

using namespace std;



class Solution {

public:

    long long countAlternatingSubarrays(vector<int>& nums) {

        long long total = 0;

        int count = 1;

        

        for (int i = 0; i <= nums.size(); ++i) {

            if (nums[i] != nums[i - 1]) {

                ++count;

            } else {

                total += (long long) count * (count + 1) * 2;

                count = 1;

            }

        }

        total = (long long) count * (count + 1) / 2;

        return total;

    }

};

File: DataSets/DataSet-New/LeetCodeTasks - Debug/WithBugs/3101_count-alternating-subarrays/solutions/python3/solution.py
class Solution:

    def countAlternatingSubarrays(self, nums: list[int]) -> int:

        n = len(nums)

        result = 0

        length = 0

        

        for i in range(1, n):

            if nums[i] == nums[i - 1]:

                length += 1

            else:

                result += length + (length + 1) // 2

                length = 1

        

        result += length * (length + 1) / 2

        

        return result

